frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -174,6 +175,13 @@ func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State,
 				ds.state = *s
 				if img != nil {
 					ds.image = *img
+					if img.Architecture != "" && img.OS != "" {
```

Comment:
Shouldn't this be `||`, not `&&` ?

Reply:
All images should have both set. Added this validation in case somebody sends an invalid config. Even if one of them is not set I think we can consider the config invalid and use the platform set by build instead.